overlord/servicestate/resources/quotas.go

```diff
@@ -0,0 +1,105 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+* Copyright (C) 2021 Canonical Ltd
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License version 3 as
+* published by the Free Software Foundation.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*
+ */
+
+package resources
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/gadget/quantity"
+)
+
+// QuotaResourceMemory is the memory limit for the quota group being controlled,
+// either the initial limit the group is created with for the "create"
+// action, or if non-zero for the "update" the memory limit, then the new
+// value to be set.
+type QuotaResourceMemory struct {
+	MemoryLimit quantity.Size
+}
+
+type QuotaResources struct {
+	Memory *QuotaResourceMemory
+}
+
+func (qr *QuotaResources) validateMemoryQuota() error {
+	// make sure the memory limit is not zero
+	if qr.Memory.MemoryLimit == 0 {
+		return fmt.Errorf("cannot create quota group with no memory limit set")
```

Comment:
This could be handled by the limit < 4kB that happens after.

Reply:
I prefer having separate error messages for the different cases